src/agreement.rs

```diff
@@ -209,7 +196,7 @@ where
     }
 }
 
-impl<B: AsRef<[u8]>> UnparsedPublicKey<B> {
```

Comment:
Will removing the `AsRef<[u8]>` bound cause any usability problems? In `src/signature.rs` a bound was added on `verify` to make up for it but nothing was added here.

Reply:
`agreement::agree_ephemeral` already has an `AsRef<[u8]>` bound so nothing needed to be added.

I think if this change were problematic, the tests would fail to compile. Maybe I'm overlooking something?

Reply:
Ah ok, I missed that it already had that bound present. I think everything is fine then in that case.